packages/astro/src/runtime/client/idle.ts

```diff
@@ -1,14 +1,22 @@
 import type { ClientDirective } from '../../@types/astro.js';
 
-const idleDirective: ClientDirective = (load) => {
+const idleDirective: ClientDirective = (load, options) => {
 	const cb = async () => {
 		const hydrate = await load();
 		await hydrate();
 	};
+
+	const rawOptions =
+		typeof options.value === 'object' ? (options.value as IdleRequestOptions) : undefined;
+
+	const idleOptions: IdleRequestOptions = {
+		timeout: rawOptions?.timeout,
+	};
```

Comment:
Since this code is shipped to all users using `client:idle`, could we simply this as this?

```js
const timeout = options?.timeout

// ...
requestIdleCallback(cb, { timeout })
setTimeoutcb, timeout || 200)
```

Reply:
I looked at the other client directives and how `rootMargin` got passed. I simply used the same code that has already been shipped.

Reply:
Let's keep it as is then and we can optimize it later